client/order/bookside_test.go

```diff
@@ -0,0 +1,831 @@
+						makeOrder([32]byte{'d'}, msgjson.BuyOrderNum, 3, 2, 10),
+						makeOrder([32]byte{'e'}, msgjson.BuyOrderNum, 1, 2, 10),
+					},
+				},
+				makeRateIndex([]uint64{1, 2}),
+				ascending,
+			),
+		},
+		{
+			side: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						makeOrder([32]byte{'a'}, msgjson.BuyOrderNum, 10, 1, 10),
+						makeOrder([32]byte{'b'}, msgjson.BuyOrderNum, 5, 1, 10),
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+			entry: makeOrder([32]byte{'c'}, msgjson.BuyOrderNum, 2, 2, 10),
+			expected: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						makeOrder([32]byte{'a'}, msgjson.BuyOrderNum, 10, 1, 10),
+						makeOrder([32]byte{'b'}, msgjson.BuyOrderNum, 5, 1, 10),
+					},
+					2: {
+						makeOrder([32]byte{'c'}, msgjson.BuyOrderNum, 2, 2, 10),
+					},
+				},
+				makeRateIndex([]uint64{1, 2}),
+				ascending,
+			),
+		},
+		{
+			side: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						makeOrder([32]byte{'a'}, msgjson.BuyOrderNum, 10, 1, 10),
+						makeOrder([32]byte{'b'}, msgjson.BuyOrderNum, 5, 1, 10),
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+			entry: makeOrder([32]byte{'c'}, msgjson.BuyOrderNum, 2, 1, 10),
+			expected: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						makeOrder([32]byte{'a'}, msgjson.BuyOrderNum, 10, 1, 10),
+						makeOrder([32]byte{'b'}, msgjson.BuyOrderNum, 5, 1, 10),
+						makeOrder([32]byte{'c'}, msgjson.BuyOrderNum, 2, 1, 10),
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+		},
+		{
+			side: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						makeOrder([32]byte{'a'}, msgjson.BuyOrderNum, 10, 1, 10),
+						makeOrder([32]byte{'c'}, msgjson.BuyOrderNum, 2, 1, 10),
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+			entry: makeOrder([32]byte{'b'}, msgjson.BuyOrderNum, 5, 1, 10),
+			expected: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						makeOrder([32]byte{'a'}, msgjson.BuyOrderNum, 10, 1, 10),
+						makeOrder([32]byte{'b'}, msgjson.BuyOrderNum, 5, 1, 10),
+						makeOrder([32]byte{'c'}, msgjson.BuyOrderNum, 2, 1, 10),
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+		},
+		{
+			side: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						makeOrder([32]byte{'a'}, msgjson.BuyOrderNum, 10, 1, 10),
+						makeOrder([32]byte{'c'}, msgjson.BuyOrderNum, 2, 1, 10),
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+			entry: makeOrder([32]byte{'b'}, msgjson.BuyOrderNum, 5, 1, 5),
+			expected: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						makeOrder([32]byte{'b'}, msgjson.BuyOrderNum, 5, 1, 5),
+						makeOrder([32]byte{'a'}, msgjson.BuyOrderNum, 10, 1, 10),
+						makeOrder([32]byte{'c'}, msgjson.BuyOrderNum, 2, 1, 10),
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+		},
+		{
+			side: makeBookSide(
```

Comment:
Comment here that this is the test checking sorting by order ID.

Reply:
noted, will label each test on what it's doing.